server.py

```diff
@@ -1,23 +1,76 @@
 import socket
+import email.utils
 
-ADDR = ('127.0.0.1', 7000)
 
-socket = socket.socket(
-    socket.AF_INET, socket.SOCK_STREAM, socket.IPPROTO_IP
-    )
+def response_ok(uri):
+    returnstr = []
+    content_type = "text/plain"
+    bytes = uri.encode('utf-8')
+    content_length = str(len(bytes))
+    date = email.utils.formatdate(usegmt=True)
+    returnstr.append("HTTP/1.1 200 OK")
+    returnstr.append("Date:" + date)
+    returnstr.append("Content-Type:" + content_type)
+    returnstr.append("Content-Length:" + content_length)
+    header = '\r\n'.join(returnstr)
+    header = header + "\r\n" + uri
+    return header
 
-socket.bind(ADDR)
-socket.listen(1)
 
-while True:
+def response_error():
+    return "500 Server Error \r\n\r\n"
+
+
+def parse_request(fullmsg):
     try:
-        conn, addr = socket.accept()
-        while True:
-            msg = conn.recv(16)
-            print msg
-            conn.sendall(msg)
-            if len(msg) < 16:
-                conn.close()
-                break
-    except KeyboardInterrupt:
-        break
\ No newline at end of file
+        tline1 = fullmsg[0]
```

Comment:
Here you are assuming that the `fullmsg` parameter you've been passed is bound to a list.  But that's because you are splitting the request *outside* this function.  The value that comes in should be the full, unaltered bytestring that arrived via your socket.  You should not process it at all until it reaches this function.  All parsing of the request should take place inside the `parse_request` function.